app/controllers/spree/products_controller_decorator.rb

```diff
@@ -5,7 +5,7 @@
   private
 
     def redirect_gift_card
-      redirect_to new_gift_card_path and return false if @product.try :is_gift_card?
+      redirect_to new_gift_card_path(product_id: @product) and return false if @product.try :is_gift_card?
```

Comment:
@nishant-cyro we need to display the product as already selected, which was selected from product listing.